backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java

```diff
@@ -211,114 +101,29 @@ public AccountResponse submitOrganizationAccountRequest(
-              request.getDefaultTestingDevice()));
-    }
-    DeviceSpecimenTypeHolder deviceSpecimenTypes =
-        _dts.getTypesForFacility(defaultTestingDeviceId, testingDeviceIds);
-
-    StreetAddress facilityAddress =
-        _avs.getValidatedAddress(
-            request.getStreetAddress1(),
-            request.getStreetAddress2(),
-            request.getCity(),
-            request.getState(),
-            request.getZip(),
-            _avs.FACILITY_DISPLAY_NAME);
-    StreetAddress providerAddress =
-        new StreetAddress(
-            Translators.parseString(request.getOpStreetAddress1()),
-            Translators.parseString(request.getOpStreetAddress2()),
-            Translators.parseString(request.getOpCity()),
-            Translators.parseState(request.getOpState()),
-            Translators.parseString(request.getOpZip()),
-            Translators.parseString(request.getOpCounty()));
-
-    PersonName providerName =
-        Translators.consolidateNameArguments(
-            null, request.getOpFirstName(), null, request.getOpLastName(), null, true);
-
-    String orgExternalId = getOrgExternalId(request.getName(), request.getState());
-
-    return _os.createOrganizationAndFacility(
-        request.getName(),
-        getTypeFromLabelOrValue(request.getType()),
-        orgExternalId,
-        request.getFacilityName(),
-        request.getCliaNumber(),
-        facilityAddress,
-        Translators.parsePhoneNumber(request.getFacilityPhoneNumber()),
-        null,
-        deviceSpecimenTypes,
-        providerName,
-        providerAddress,
-        Translators.parsePhoneNumber(request.getOpPhoneNumber()),
-        request.getNpi());
-  }
-
   private Organization checkAccountRequestAndCreateOrg(OrganizationAccountRequest request) {
-    String organizationName = request.getName();
-
     // verify that the organization doesn't already exist
-    checkForDuplicateOrg(organizationName);
-
-    String orgExternalId = getOrgExternalId(organizationName, request.getState());
-
-    String organizationType = getTypeFromLabelOrValue(request.getType());
+    String organizationName = checkForDuplicateOrg(request.getName(), request.getState());
+    String orgExternalId = createOrgExternalId(organizationName, request.getState());
+    String organizationType = Translators.parseOrganizationType(request.getType());
     return _os.createOrganization(organizationName, organizationType, orgExternalId);
   }
 
-  private void checkForDuplicateOrg(String organizationName) {
+  private String checkForDuplicateOrg(String organizationName, String state) {
     Optional<Organization> potentialDuplicateOrg = _os.getOrganizationByName(organizationName);
     if (potentialDuplicateOrg.isPresent()) {
-      throw new BadRequestException("Organization is a duplicate.");
+      // potential duplicate orgs must be in the same state to be a true duplicate
+      if (potentialDuplicateOrg.get().getExternalId().startsWith(state)) {
```

Comment:
FYI I think there are some `organizations` with out states in their external ID.

Reply:
Yeah, I was worried that might be the case with older orgs. We don't really have a way to check the org's state in that case, so I just added a catch for the `UnexpectedRollbackException` that re-throws it as a BadRequest. It won't allow duplicate org names across states, but at least users will get a toast explaining what happened.